core/sourcehook/sh_memory.h

```diff
@@ -67,27 +67,28 @@ namespace SourceHook
 			// 08048000-0804c000 r-xp 00000000 03:03 1010107    /bin/cat
 			unsigned long rlower, rupper;
 			char r, w, x;
-			while (fscanf(pF, "%lx-%lx %c%c%c", &rlower, &rupper, &r, &w, &x) != EOF) {
+			static char *buffer = NULL;
```

Comment:
this is incorrect - it shouldn't be static.

Reply:
neither of them should be*

Reply:
Getline will resize with realloc as needed, shouldn't these be static so we don't need to allocate more times than necessary?

Reply:
I'm confused, this is a stack variable? making it static moves it to a shared single instance for every function call, and multiple calls to this function will now share that variable and corrupt eachother. SH in general isn't threadsafe, but the present proposal will result in memory corruption.

Reply:
Oh true I wasn't really thinking about that. I'd like to avoid having memory alloced and freed every call but I'm not sure if there is another way to do it.

Reply:
Made another benchmark with non static buffer, It's pretty negligible (alloc actually beat static once) so I'll make the change.
Results (3 runs):
```
Elapsed CPU Time (STATIC) = '9.18869' sec.
Elapsed CPU Time per Iteration (STATIC, 500000) = '1.837737e-05' sec.
Elapsed CPU Time (ALLOC) = '9.30199' sec.
Elapsed CPU Time per Iteration (ALLOC, 500000) = '1.860398e-05' sec.

Elapsed CPU Time (STATIC) = '9.09555' sec.
Elapsed CPU Time per Iteration (STATIC, 500000) = '1.819110e-05' sec.
Elapsed CPU Time (ALLOC) = '9.16725' sec.
Elapsed CPU Time per Iteration (ALLOC, 500000) = '1.833449e-05' sec.

Elapsed CPU Time (STATIC) = '9.10335' sec.
Elapsed CPU Time per Iteration (STATIC, 500000) = '1.820671e-05' sec.
Elapsed CPU Time (ALLOC) = '9.09759' sec.
Elapsed CPU Time per Iteration (ALLOC, 500000) = '1.819519e-05' sec.
```